### PR TITLE
Add features and fix bugs in Quickstart for parity with Java

### PR DESCRIPTION
**Pull request by yihezkel**
#### Pull Request Description

Add features and fix bugs in Quickstart for parity with Java

---

#### Future Release Comment
_[Add description of your change, to include in the next release]_
_[Delete any or all irrelevant sections, e.g. if your change does not warrant a release comment at all]_

**Features:**
- Add features to Quickstart for parity with Java

**Fixes:**
- Fix bugs in Quickstart

### CODE REVIEW

File: quick_start/kusto_sample_app.py [hunk 1]
```diff
@@ -20,394 +16,458 @@
...
-        # Learn More: For additional information about supported data formats, see
-        #  https://docs.microsoft.com/azure/data-explorer/ingestion-supported-formats
-        data_format=file_format,
-        # Todo - Config: Setting the ingestion batching policy takes up to 5 minutes to have an effect.
-        #  For the sake of the sample we set Flush-Immediately, but in practice it should not be commonly used.
-        #  Comment the below line after running the sample for the first few times!
-        flush_immediately=True,
-    )
+    @classmethod
+    def ingest(cls, file: dict, data_format: DataFormat, ingest_client: QueuedIngestClient, database_name: str, table_name: str, mapping_name: str) -> None:
```
Comment (yihezkel) [inline quick_start/kusto_sample_app.py:320]: Prefer a parent of `QueuedIngestClient`, which doesn't exist. Asaf is adding one, so we should change this once it merges.

File: quick_start/kusto_sample_app.py [hunk 2]
```diff
@@ -20,394 +16,458 @@
...
+        #  See the SDK's samples and the E2E tests in azure.kusto.ingest for additional references.
+        if source_type == "localfilesource":
+            cls.ingest_from_file(ingest_client, database_name, table_name, uri, data_format, mapping_name)
+        elif source_type == "blobsource":
+            cls.ingest_from_blob(ingest_client, database_name, table_name, uri, data_format, mapping_name)
+        else:
+            print(f"Unknown source '{source_type}' for file '{uri}'")
+
+    @classmethod
+    def ingest_from_file(cls, ingest_client: QueuedIngestClient, database_name: str, table_name: str, file_path: str,
```
Comment (yihezkel) [inline quick_start/kusto_sample_app.py:338]: Prefer a parent of `QueuedIngestClient`, which doesn't exist. Asaf is adding one, so we should change this once it merges.

File: quick_start/kusto_sample_app.py [hunk 3]
```diff
@@ -20,394 +16,458 @@
...
+        ingestion_properties = cls.create_ingestion_properties(database_name, table_name, data_format, mapping_name)
+
+        # Tip 1: For optimal ingestion batching and performance, specify the uncompressed data size in the file descriptor instead of the default below of 0.
+        #  Otherwise, the service will determine the file size, requiring an additional s2s call, and may not be accurate for compressed files.
+        # Tip 2: To correlate between ingestion operations in your applications and Kusto, set the source ID and log it somewhere
+        file_descriptor = FileDescriptor(file_path, size=0, source_id=uuid.uuid4())
+        ingest_client.ingest_from_file(file_descriptor, ingestion_properties=ingestion_properties)
+
+    @classmethod
+    def ingest_from_blob(cls, client: QueuedIngestClient, database_name: str, table_name: str, blob_url: str,
```
Comment (yihezkel) [inline quick_start/kusto_sample_app.py:349]: Prefer a parent of `QueuedIngestClient`, which doesn't exist. Asaf is adding one, so we should change this once it merges.

File: quick_start/oneclick_instruction_box.md [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
-5. Run `python kusto_sample_app.py`. It will already be configured to your cluster and source.
+1. Download the **DOWNLOAD_LINK** as a ZIP file.
+1. Extract the app source code.
+1. Open a command line window and navigate to the folder where you extracted the script.
+1. Run `python -m pip install azure-kusto-data azure-kusto-ingest`.
+1. Run `python kusto_sample_app.py`. It will already be configured to your cluster and source.
 
 ### Troubleshooting
 
-* If you are having trouble running python on your machine or need instructions on how to install python, you can consult the following [python environment setup tutorial](https://www.tutorialspoint.com/python/python_environment.htm) from _Tutorials Point_.
```
Comment (yogilad) [inline quick_start/oneclick_instruction_box.md:11]: Why did you delete the first bullet?

**Review by yogilad — APPROVED** [review]

File: quick_start/oneclick_instruction_box.md (hunk 4 above)
Comment (yihezkel) [reply quick_start/oneclick_instruction_box.md:11]: It's just moved up and merged with "Prerequisites", per Shlomo's request
